### PR TITLE
support multi-cards magnitude pruning

### PR DESCRIPTION
**Pull request by lkk12014402**
## Type of Change

API not changed

## Description

support multi-cards magnitude pruning

### CODE REVIEW

**Comment by hshen14** [comment]
Add a readme to show how to use it.

File: examples/pytorch/nlp/huggingface_models/language-modeling/pruning/magnitude/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+Step-by-Step
+============
+
+# single GPU
+
+```
+bash run.sh
```
Comment (hshen14) [inline examples/pytorch/nlp/huggingface_models/language-modeling/pruning/magnitude/README.md:7]: Suggest to expose the model name here

File: examples/pytorch/nlp/huggingface_models/language-modeling/pruning/magnitude/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      "warmup_type": "cosine"
+    }
+  }
+}
+```
+
+### pruning
+
+```
+bash run_ds.sh
```
Comment (hshen14) [inline examples/pytorch/nlp/huggingface_models/language-modeling/pruning/magnitude/README.md:89]: same as above

File: examples/pytorch/nlp/huggingface_models/language-modeling/pruning/magnitude/README.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+```
+bash run.sh
+```
+
+# multi GPU
+
+we use `accelerate` and `deepspeed ZeRO Stage-2` to conduct weight magnitude pruning
+
+### Accelerate DeepSpeed Plugin
```
Comment (hshen14) [inline examples/pytorch/nlp/huggingface_models/language-modeling/pruning/magnitude/README.md:14]: where to specify the number of GPU cards?

Comment (lkk12014402) [reply examples/pytorch/nlp/huggingface_models/language-modeling/pruning/magnitude/README.md:14]: accelerate config commands will ask you to set the gpu cards

**Review by hshen14 — APPROVED** [review]

File: examples/pytorch/nlp/huggingface_models/language-modeling/pruning/magnitude/README.md (hunk 2 above)
Comment (lkk12014402) [reply examples/pytorch/nlp/huggingface_models/language-modeling/pruning/magnitude/README.md:89]: they are different. the above one is use for accelerate which is auto generated by 'accelerate config', this one is only for deepspeed which path should be set in the above file

File: examples/pytorch/nlp/huggingface_models/language-modeling/pruning/magnitude/run_ds.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    --dataset_name NeelNanda/pile-10k \
+    --model_name_or_path facebook/opt-125m \
+    --block_size 128 \
+    --per_device_train_batch_size 8 \
+    --gradient_accumulation_steps 4 \
+    --output_dir ./test-clm \
+    --num_train_epochs 1 \
+    --max_train_steps 100 \
+    --do_prune \
+    --num_train_epochs 10 \
```
Comment (wenhuach21) [inline examples/pytorch/nlp/huggingface_models/language-modeling/pruning/magnitude/run_ds.sh:14]: two num_train_epochs.  what the difference between num_train_epochs and max_train_steps. how about just using train_steps

**Review by wenhuach21 — APPROVED** [review]

File: examples/pytorch/nlp/huggingface_models/language-modeling/pruning/magnitude/run_ds.sh (hunk 4 above)
Comment (lkk12014402) [reply examples/pytorch/nlp/huggingface_models/language-modeling/pruning/magnitude/run_ds.sh]: fixed

**Pull request merged**